specs/OAuth/OAuth2Manager.md

```diff
@@ -0,0 +1,956 @@
+
+## TokenResponse Properties
+
+| Name | Description | Type |
+|-|-|-|
+| AccessToken | Derived from the "access_token" parameter of the token response. | String |
+| TokenType | Derived from the "token_type" parameter of the token response. | String |
+| ExpiresIn | Derived from the "expires_in" parameter of the token response.  It's the lieftime of access token in seconds. | String |
+| RefreshToken | Derived from the "refresh_token" parameter of the token response. | String |
+| Scope | Derived from the "scope" parameter of the token response. | String |
+| AdditionalParams | Additional parameters set by the authorization server in the response URI. | Windows.Foundation.Collections.IMapView<String, String> |
+
+## TokenFailureKind enum
+
+It's an enum that specifies the kind of token failure.
+
+| Name | Description |
+|-|-|
+| ErrorResponse | The server responded with an error response as described by RFC 6749 section 5.2. This means that the failure object has an 'Error' string and possibly other specified properties |
+| HttpFailure | The HTTP POST request failed. See the 'ErrorCode' property for more details as to why. |
+| InvalidResponse | The server responded, but its response was improperly formatted. This could be that the server did not send the response as JSON, the response JSON string was improperly formatted, or the response JSON contained unexpected object types (e.g. a number when a string is expected, etc.). |
+
+## TokenFailure class
+
+It's a class that provides the failure object of a token request.
+
+## TokenFailure Properties
+
+| Name | Description | Type |
+|-|-|-|
+| Kind | Specifies the kind of token failure. | TokenFailureKind |
+| Error | Derived from the "error" parameter of the error response. | String |
+| ErrorDescription | Derived from the "error_description" parameter of the error response. | String |
+| ErrorUri | Derived from the "error_uri" parameter of the error response. | Windows.Foundation.Uri |
+| ErrorCode | The HTTP error code that was returned. | Windows.Web.Http.HttpStatusCode |
+| AdditionalParams | Additional parameters set by the authorization server in the response URI. | Windows.Foundation.Collections.IMapView<String, String> |
+
+
+# API Details
+
+> Note: all of this new WinAppSDK API is to support 
+  OAuth 2.0 launch using browser launch.
+ 
+```c++ (but really MIDL3)
+namespace Microsoft.Security.Authentication.OAuth
+{
+    [contractversion(1)]
+    apicontract OAuthContract {};
+
+    [contract(OAuthContract, 1)]
+    runtimeclass ClientAuthentication
+    {
+        ClientAuthentication();
+        ClientAuthentication(Windows.Web.Http.Headers.HttpCredentialsHeaderValue authorization);
+
+        static ClientAuthentication CreateForBasicAuthorization(String clientId, String clientSecret);
+
+        // Specifies the 'Authorization' header of the HTTP POST request when requesting a token
+        Windows.Web.Http.Headers.HttpCredentialsHeaderValue Authorization { get; set; };
```

Comment:
Nit on MIDL3 style - for read-write parameters, use the "compact" style of properties:

```c#
runtimeclass Foo {
    String Bar;
}
```